server/main.go

```diff
@@ -24,15 +24,29 @@ import (
 	"github.com/tigrisdata/tigris/store/search"
 	"github.com/tigrisdata/tigris/util"
 	ulog "github.com/tigrisdata/tigris/util/log"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
 )
 
+func initTracer() {
+	if !config.DefaultConfig.Tracing.Enabled {
+		return
+	}
+	tracer.Start(
+		tracer.WithUDS("/var/run/datadog/apm.socket"),
+	)
+	defer tracer.Stop()
```

Comment:
this won't work because as the function return this tracer will stop, we need to return this tracer from this method and add this defer in main.

Reply:
I surmised that might not. Ok, let me do that!